tests/watcher_test.go

```diff
@@ -0,0 +1,127 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	client "k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+)
+
+const (
+	TEST_GLOBAL_SVC_NAME = "echo-test-svc-global"
+	TEST_NS              = "default"
+	TEST_RETRIES         = 5
+)
+
+func GetEndpintSlices() {
+	var kubeconfig *string
+	if home := homedir.HomeDir(); home != "" {
+		kubeconfig = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
+	}
+
+	//use the current context in kubeconfig
+	config, err := client.BuildConfigFromFlags("", *kubeconfig)
+
+	fmt.Errorf(err.Error())
+	//Make sure config is not nil
+
+	client, err := kubernetes.NewForConfig(config)
+
+	//Make sure global-svc is created.
+	_, err = client.CoreV1().Services(TEST_NS).Get(context.Background(), TEST_GLOBAL_SVC_NAME, metav1.GetOptions{})
+
+}
+
+func TestGlobalService(t *testing.T) {
+	t.Logf("Testing if Service: %v is created", TEST_GLOBAL_SVC_NAME)
+
+	// Make request && make sure we get response from both cluster in X retries.
+	// Add retrieved responses inside map of string and make sure length is 2.
+	cluster := make(map[string]string)
+	tries := 0
+	for tries < TEST_RETRIES {
+		tries += 1
+		// Try service for couple of times
+		url := fmt.Sprintf("http://%v.%v", TEST_GLOBAL_SVC_NAME, TEST_NS)
+
+		resp, err := http.Get(url)
+
+		assert.NoError(t, err, "Unable to make HTTP request")
+		defer resp.Body.Close()
+
+		assert.Equal(t, resp.StatusCode, http.StatusOK)
+
+		body, err := ioutil.ReadAll(resp.Body)
+		assert.NoError(t, err, "Unable to read Response body")
+
+		respBody := string(body)
+		cluster_name := strings.Split(respBody, "Hello from: ")[1]
```

Comment:
Checking a 200 is probably enough? Tests shouldn't contain a lot of logic to do any parsing or things like that. What are we testing here? Connectivity? The echo server?

Reply:
Also, a testing tool is not great if it requires us to do a bunch of parsing. Why do we need a response that says `Hello from: <something>`. I'd prefer to just have `<something>` returned.

Reply:
Hey while I agree testing server shouldnt have lot of logic,  this echo server is not written specifically for this. It echoes whatever information is set in `env` variable with String `Hello from: <env_variable_passed>` for this purpose we need to string manipulation. Other way to check would have using String contains option.

Please let me know if there is another service i should used. I looked at `bb` but I thought it will unnecessarily complicate the setup as it has one `grpc` service and `http` service, and both of them need to run together from my understanding. As we are focusing simply on `http` for this tests, I thought it will be quick and easier.